xbmc/filesystem/BlurayDirectory.cpp

```diff
@@ -180,35 +671,51 @@ void CBlurayDirectory::GetTitles(bool main, CFileItemList &items)
   }
 }
 
-void CBlurayDirectory::GetRoot(CFileItemList &items)
+void CBlurayDirectory::GetRoot(CFileItemList& items)
 {
-    GetTitles(true, items);
+  GetTitles(true, items);
 
-    CURL path(m_url);
-    CFileItemPtr item;
-
-    path.SetFileName(URIUtils::AddFileToFolder(m_url.GetFileName(), "titles"));
-    item = std::make_shared<CFileItem>();
-    item->SetPath(path.Get());
-    item->m_bIsFolder = true;
-    item->SetLabel(g_localizeStrings.Get(25002) /* All titles */);
-    item->SetArt("icon", "DefaultVideoPlaylists.png");
-    items.Add(item);
-
-    const BLURAY_DISC_INFO* disc_info = bd_get_disc_info(m_bd);
-    if (disc_info && disc_info->no_menu_support)
-    {
-      CLog::Log(LOGDEBUG, "CBlurayDirectory::GetRoot - no menu support, skipping menu entry");
-      return;
-    }
+  AddRootOptions(items);
+}
+
+void CBlurayDirectory::GetRoot(CFileItemList& items,
+                               const CFileItem& episode,
+                               const std::vector<CVideoInfoTag>& episodesOnDisc)
+{
+  GetEpisodeTitles(episode, items, episodesOnDisc);
 
-    path.SetFileName("menu");
-    item = std::make_shared<CFileItem>();
-    item->SetPath(path.Get());
-    item->m_bIsFolder = false;
-    item->SetLabel(g_localizeStrings.Get(25003) /* Menus */);
-    item->SetArt("icon", "DefaultProgram.png");
-    items.Add(item);
+  if (!items.IsEmpty())
+    AddRootOptions(items);
+}
+
+void CBlurayDirectory::AddRootOptions(CFileItemList& items)
+{
+
+  CURL path(m_url);
+  CFileItemPtr item;
+
+  path.SetFileName(URIUtils::AddFileToFolder(m_url.GetFileName(), "titles"));
+  item = std::make_shared<CFileItem>();
+  item->SetPath(path.Get());
+  item->m_bIsFolder = true;
```

Comment:
Remove declaration of `item`in line 695 and condense three lines to one
`const auto item{std::make_shared<CFileItem>(path.Get(), true)};`